nupic/research/frameworks/dendrites/modules/dendritic_mlp.py

```diff
@@ -159,21 +162,32 @@ def __init__(
 
             input_size = self.hidden_sizes[i]
 
-        self._output_layer = nn.Sequential()
-        output_linear = SparseWeights(module=nn.Linear(input_size, output_size),
-                                      sparsity=weight_sparsity, allow_extremes=True)
-        if weight_init == "modified":
-            self._init_sparse_weights(output_linear, 1 - kw_percent_on if kw else 0.0)
-        self._output_layer.add_module("output_linear", output_linear)
+        self._single_output_head = not isinstance(output_size, Iterable)
+        if self._single_output_head:
+            output_size = (output_size,)
+
+        self._output_layers = nn.ModuleList()
+        for out_size in output_size:
+            output_layer = nn.Sequential()
+            output_linear = SparseWeights(module=nn.Linear(input_size, out_size),
```

Comment:
shouldn't the size of the output layer be `(hidden_sizes[-1], out_size)`?

Reply:
input_size is updated in line 163 to be the last hidden size, so it should be the same.